src/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCase.php

```diff
@@ -4,17 +4,48 @@
 
 namespace WMDE\Fundraising\Frontend\UseCases\HandlePayPalPaymentNotification;
 
+use WMDE\Fundraising\Frontend\Domain\Model\Donation;
+use WMDE\Fundraising\Frontend\Domain\Repositories\DonationRepository;
+use WMDE\Fundraising\Frontend\Domain\Repositories\GetDonationException;
+use WMDE\Fundraising\Frontend\Infrastructure\DonationAuthorizer;
+
 /**
  * @license GNU GPL v2+
  * @author Kai Nissen < kai.nissen@wikimedia.de >
  */
 class HandlePayPalPaymentNotificationUseCase {
 
-	public function __construct() {
+	private $repository;
+	private $authorizationService;
+
+	public function __construct( DonationRepository $repository, DonationAuthorizer $authorizationService ) {
+		$this->repository = $repository;
+		$this->authorizationService = $authorizationService;
 	}
 
-	public function handleNotification( HandlePayPalPaymentNotificationRequest $request ) {
-		return 'TODO';
+	public function handleNotification( PayPalNotificationRequest $request ) {
```

Comment:
I think so. PayPal wouldn't process the response body, but only the status code.